Implement remote contract events support

davidyuk opened the pull request:
depends on https://github.com/aeternity/aepp-calldata-js/pull/112
depends on #1395 
closes #1344 except for events with the same name in different contracts

marc0olo commented:
that's awesome! =)

do we have any idea to handle same names?

I didn't check in detail yet how it works with this PR. do we rely on any additional http request or is it fully covered with calldata-lib in the sdk?

davidyuk commented:
> do we have any idea to handle same names?

nope

> do we rely on any additional http request or is it fully covered with calldata-lib in the sdk?

the events before were decoded by sdk, without requests to http compiler or so, in this PR I've removed our poor implementation and switched to calldata

thepiwo commented on test/integration/contract-aci.js:
```diff
@@ -520,7 +543,7 @@ describe('Contract instance', function () {
       it('Invalid arguments length', async () => {
         await expect(testContract.methods.datTypeGFn())
-          .to.be.rejectedWith('Non matching number of arguments. Got 0 but expected from 1 to 1')
+          .to.be.rejectedWith('Non matching number of arguments. datTypeGFn expects between 1 and 1 number of arguments but got 0')
```
weird error message

marc0olo replied:
@davidyuk does this message come from calldata-lib ? just wondering

thepiwo reviewed (approved):
pretty good if I understand correctly

davidyuk commented:
> do we have any idea to handle same names?

I've added `contractAddressToName` option to resolve this

marc0olo reviewed (approved):
lgtm
